packages/core/admin/admin/src/content-manager/hooks/useRelation/useRelation.js

```diff
@@ -56,28 +56,30 @@ export const useRelation = (cacheKey, { relation, search }) => {
     },
     select: (data) => ({
       ...data,
-      pages: data.pages.map((page) => {
-        if (!page) {
-          return page;
-        }
-
-        const { data, results, pagination } = page;
-        const isXToOneRelation = !!data;
-        let normalizedResults = [];
-
-        // xToOne relations return an object, which we normalize so that relations
-        // always have the same shape
-        if (isXToOneRelation) {
-          normalizedResults = [data];
-        } else if (results) {
-          normalizedResults = results.reverse();
-        }
-
-        return {
-          pagination,
-          results: normalizedResults,
-        };
-      }),
+      pages: data.pages
+        .map((page) => {
+          if (!page) {
+            return page;
+          }
+
+          const { data, results, pagination } = page;
+          const isXToOneRelation = !!data;
+          let normalizedResults = [];
+
+          // xToOne relations return an object, which we normalize so that relations
+          // always have the same shape
+          if (isXToOneRelation) {
+            normalizedResults = [data];
+          } else if (results) {
+            normalizedResults = [...results].reverse();
+          }
+
+          return {
+            pagination,
+            results: normalizedResults,
+          };
+        })
+        .reverse(),
```

Comment:
Am i being dumb, how come we reverse it here and on L74?

Reply:
I don't understand it either and even less, why no test case covers the problem. I'd like to work on this early next week, because we had to change this a couple of times now and the number of `.reverse()` calls is getting out of hand.

Reply:
You can test the order on the main branch right now, it is not working. I think it doesn't work because reverse acts weirdly by mutating, that's why the proposition of fix is to spread the array first.

Previously we only handled order in `normalizeRelations`.
Not long ago we decided to do the inner array reverse during the fetching phase in the select method. But we kept the outer reverse in the `normalizeRelations`. Which I felt didn't make sense to have 2 places to handle this.

To give context @joshuaellis we receive
```
[{id: 1}, {id:2}]
[{id: 3}, {id:4}]
```

And we want

```
[{id: 4}, {id:3}]
[{id: 2}, {id:1}]
```

So inner reverse then outer reverse

I agree this needs some test coverage

Reply:
fixed `normalizeRelations` tests (which is not handling ordering anymore) and added tests for `useRelation` (which is now the only one to handle ordering) [here](https://github.com/strapi/strapi/pull/14631/commits/0f6e26742fbf6605d3448c791a54b7c13fdbe5c0)
let me know what you think about the fix/refactor